From 9414d7a122c50f7834a5783d5ad69b9635870fe2 Mon Sep 17 00:00:00 2001
From: Jonathan David <jonathan.david@ni.com>
Date: Tue, 15 Mar 2016 14:13:28 -0500
Subject: [PATCH] Added curl_formadd_va() API function

Added API function with `va_list` variable argument parameter in
addition to cURL's function using `...`.

Upstream-Status: Denied

This patchset was denied by upstream because they did not agree that our
usecase for these added functions were valid - and we did not have a
defense.

https://curl.se/mail/lib-2017-01/0146.html

Signed-off-by: Alex Stewart <alex.stewart@ni.com>
---
 lib/formdata.c | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/lib/formdata.c b/lib/formdata.c
index e40c4bc..11f142f 100644
--- a/lib/formdata.c
+++ b/lib/formdata.c
@@ -690,6 +690,13 @@ CURLFORMcode FormAdd(struct curl_httppost **httppost,
  * @unittest: 1308
  */
 
+CURLFORMcode curl_formadd_va(struct curl_httppost **httppost,
+                             struct curl_httppost **last_post,
+                             va_list arg)
+{
+  return FormAdd(httppost, last_post, arg);
+}
+
 CURLFORMcode curl_formadd(struct curl_httppost **httppost,
                           struct curl_httppost **last_post,
                           ...)
@@ -697,7 +704,7 @@ CURLFORMcode curl_formadd(struct curl_httppost **httppost,
   va_list arg;
   CURLFORMcode result;
   va_start(arg, last_post);
-  result = FormAdd(httppost, last_post, arg);
+  result = curl_formadd_va(httppost, last_post, arg);
   va_end(arg);
   return result;
 }
